File: Java/src/ch01/CountBitOne.java
package ch01;

/**
 * 统计二进制中一的个数
 */
public class CountBitOne {

    public static void main(String[] args) {
        System.out.println(countBitOne(0));
        System.out.println(countBitOne(1));
        System.out.println(countBitOne(2));
        System.out.println(countBitOne(3));
    }

    /**
     * 统计二进制中一的个数
     * @param N 给定的一个数
     * @return 给定的数中，转换成二进制的时候，1的个数
     */
    public static int countBitOne(int N){
        int count = 0;
        while (N != 0){
            int rightOne = N & ((~N) + 1);
            count++;
            N = N ^ rightOne;
        }
        return count;
    }
}

File: Python/bak/ch01/CountBitOne.py
def countBitOne(n):
    """
    统计二进制中一的个数
    :param n: 给定的一个数
    :return: 给定的数，转换成二进制之后，一的个数
    """
    count = 0
    while n != 0:
        rightOne = n & ((~n) + 1)
        count = count + 1
        n = n ^ rightOne
    return count


if __name__ == '__main__':
    print(countBitOne(0))
    print(countBitOne(1))
    print(countBitOne(2))
    print(countBitOne(3))
